diff --git a/hadrian/src/Flavour.hs b/hadrian/src/Flavour.hs
index 3aabd263ce..f3e8c0f734 100644
--- a/hadrian/src/Flavour.hs
+++ b/hadrian/src/Flavour.hs
@@ -144,6 +144,7 @@ werror =
           , arg "-optc-Wno-error=unknown-pragmas"
             -- rejected inlinings are highly dependent upon toolchain and way
           , arg "-optc-Wno-error=inline"
+          , arg "-optc-Wno-error=cpp"
           ]
       -- N.B. We currently don't build the boot libraries' C sources with -Werror
       -- as this tends to be a portability nightmare.
